cmd/lassie/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	lcli "github.com/filecoin-project/lassie/cli"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/urfave/cli/v2"
+)
+
+var log = logging.Logger("lassie")
+
+func main() {
+	// set up a context that is canceled when a command is interrupted
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// set up a signal handler to cancel the context
+	go func() {
+		interrupt := make(chan os.Signal, 1)
+		signal.Notify(interrupt, syscall.SIGTERM, syscall.SIGINT)
+
+		select {
+		case <-interrupt:
+			fmt.Println()
+			log.Info("received interrupt signal")
+			cancel()
+		case <-ctx.Done():
+		}
+
+		// Allow any further SIGTERM or SIGINT to kill process
+		signal.Stop(interrupt)
+	}()
+
+	app := &cli.App{
+		Name:    "lassie",
+		Usage:   "Utility for retrieving content from the Filecoin network",
+		Suggest: true,
+		Flags: []cli.Flag{
+			lcli.FlagVerbose,
+		},
+		Commands: []*cli.Command{
+			daemonCmd,
+			fetchCmd,
+			versionCmd,
+		},
+	}
+
+	if err := app.RunContext(ctx, os.Args); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func before(cctx *cli.Context) error {
```

Comment:
neat, didn't know you could do this

what kinds of logs are we capturing with this that we wouldn't be by doing it locally, or is this simply to be able to do it in one place rather than multiple?

Reply:
I'm not sure I understand your question here. `before()` is then used as the `urfave/cli` `Command.Before` property to run before each command. The verbose flags are then added to the command's flag list. This is to ensure that all the different logging subsystems are set the appropriate level determined by the verbose flag.